16.0 - [REF] account: Adding unittests for concurrency issues in account_move sequences

moylop260 opened the pull request:
Related to https://github.com/odoo/odoo/issues/90465

Reproducing all the new cases of concurrency for Odoo v14.0 account.move records in the following cases:
- Creating draft invoices
- Creating payments
- Deadlock payment vs invoice
- Editing last invoice and creating new one
- Editing last payment and creating new one
- Reconciling last invoice and creating new one
- Reconciling last payment and creating new one
- No be able to configure to standard sequences anymore

All these kind of errors are raising other kind of errors related to subscription in payment exception, sale order in draft but with transaction, pos order errors, ecommerce users raising errors after pay then request the pay again and again and so on

All workers are used waiting for releases in the queue

Issue describing this:
  - https://github.com/odoo/odoo/issues/91873

Foward-Porting:
  - https://github.com/odoo/odoo/pull/91525


Applying the commits from:
 - https://github.com/odoo/odoo/pull/104606

robodoo commented:
[Pull request status dashboard](https://mergebot.odoo.com/odoo/odoo/pull/104647).

moylop260 commented:
Fixed for >=v16.0 from https://github.com/odoo/odoo/pull/104606
Fixed for >=14.0 and <16.0 from OCA module [account_move_name_sequence](https://github.com/OCA/account-financial-tools/tree/14.0/account_move_name_sequence)

Check the following thread to decide the future of the `account_move_name_sequence` module since that Odoo v16.0 doesn't allow to configure journals as Gaps-Allowed:
 - https://github.com/OCA/account-financial-tools/pull/1510
